llvm/lib/Transforms/Instrumentation/MemorySanitizer.cpp

```diff
@@ -282,6 +282,12 @@ static cl::opt<bool> ClPoisonUndefVectors(
              "unaffected by this flag (see -msan-poison-undef)."),
     cl::Hidden, cl::init(false));
 
+static cl::opt<bool> ClPreciseDisjointOr(
```

Comment:
why do we need a flag?

Reply:
It fixes a false negative, hence some existing tests that currently pass will start failing when the flag is turned on. Having the flag allows users to:
- turn the flag on, to preemptively identify and fix such issues while the flag is still defaulted off
- have an escape hatch for when the flag is eventually defaulted on in a future patch